osu.Game/Screens/Edit/Screens/Setup/Components/LabelledComponents/LabelledTextBox.cs

```diff
@@ -0,0 +1,195 @@
+
+        private const float label_container_width = 150;
+        private const float outer_corner_radius = 15;
+        private const float inner_corner_radius = 10;
+        private const float default_height = 40;
+        private const float default_label_left_padding = 15;
+        private const float default_label_top_padding = 12;
+        private const float default_label_text_size = 16;
+
+        public event OnCommitHandler OnCommit;
+
+        private bool readOnly;
+        public bool ReadOnly
+        {
+            get => readOnly;
+            set
+            {
+                textBox.ReadOnly = value;
+                readOnly = value;
+            }
+        }
+
+        private string labelText;
+        public string LabelText
+        {
+            get => labelText;
+            set
+            {
+                labelText = value;
+                label.Text = value;
+            }
+        }
+
+        private float labelTextSize;
+        public float LabelTextSize
+        {
+            get => labelTextSize;
+            set
+            {
+                labelTextSize = value;
+                label.TextSize = value;
+            }
+        }
+
+        private string placeholderText;
+        public string PlaceholderText
+        {
+            get => placeholderText;
+            set
+            {
+                placeholderText = value;
+                textBox.PlaceholderText = value;
+            }
+        }
+
+        private string text;
+        public string Text
+        {
+            get => text;
+            set
+            {
+                text = value;
+                textBox.Text = value;
+            }
+        }
+
+        private float height = default_height;
+        public float Height
+        {
+            get => height;
+            private set
+            {
+                height = value;
+                textBox.Height = value;
+                content.Height = value;
+            }
+        }
+
+        public MarginPadding Padding
+        {
+            get => base.Padding;
+            set
+            {
+                base.Padding = value;
+                base.Height = Height + base.Padding.Top;
+            }
+        }
+
+        public MarginPadding LabelPadding
+        {
+            get => label.Padding;
+            set => label.Padding = value;
+        }
+
+        public MarginPadding TextBoxPadding
+        {
+            get => textBox.Padding;
+            set => textBox.Padding = value;
+        }
+
+        public Color4 LabelTextColour
+        {
+            get => label.Colour;
+            set => label.Colour = value;
+        }
+
+        public Color4 BackgroundColour
+        {
+            get => content.Colour;
+            set => content.Colour = value;
+        }
+
+        public LabelledTextBox()
+        {
+            RelativeSizeAxes = Axes.X;
+            base.Height = default_height + Padding.Top;
```

Comment:
This can't happen. I don't see a reason why this is required.

This should be basically `Height = default_height;` and that's it.